### PR TITLE
[docs] development environment  setup english doc

### PR DESCRIPTION
**Pull request by Tianqi-Dotes**
create developer docs | development environment setup English doc file

### CODE REVIEW

**Comment by CalvinKirs** [comment]
Hi, @Segun-Ogundipe  can you review this PR? Its Chinese document is https://dolphinscheduler.apache.org/zh-cn/docs/development/development-environment-setup.html

**Comment by Segun-Ogundipe** [comment]
Will do @CalvinKirs

**Comment by Segun-Ogundipe** [comment]
Hi @Tianqi-Dotes, you did a good job here.
One thing though, the file is not well formatted. Follow the format in the Chinese version.
For instance, the first line should be `### Preparation` not `Preparation`

**Comment by dailidong** [comment]
> create developer docs | development environment setup English doc file
thanks for your first contribution

by the way, @Tianqi-Dotes @Segun-Ogundipe  could you setup the DolphinScheduler development successfully if you follow the this doc?  I want to know where this document needs to be improved？

if you want to have a try to setup the development environment,  please let me know

**Comment by Segun-Ogundipe** [comment]
Hi @dailidong, I'll like to give it a try

**Comment by Tianqi-Dotes** [comment]
@Segun-Ogundipe 
hi Segun,
thanx for your doc review,
will check the format part of the file according to other doc files.
have a nice day
Tianqi

**Comment by Tianqi-Dotes** [comment]
@Segun-Ogundipe 
hi Segun,
Have done the modification, please help me check the .md file again.
Deep thanx!
Tianqi

File: docs/en-us/development/development-environment-setup.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+2. Run command  
+npm install
+
+#### Set up the back-end
+
+1. Import the project to IDEA  
+file-->open
+
+2. Modify the database configurations, the 'datasource.properties' under the resource folder which belongs to the dao module
```
Comment (Segun-Ogundipe) [inline docs/en-us/development/development-environment-setup.md:72]: 2. Modify the database configuration in the datasource.properties file in the resource directory of the dao module

File: docs/en-us/development/development-environment-setup.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    mysql> flush privileges;
+```
+
+#### Set up the front-end
+
+1. Enter the dolphinscheduler-ui folder  
+cd dolphinscheduler-ui
+
+2. Run command  
+npm install
```
Comment (Segun-Ogundipe) [inline docs/en-us/development/development-environment-setup.md:65]: 2. Run `npm install`

File: docs/en-us/development/development-environment-setup.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+2. Modify the database configurations, the 'datasource.properties' under the resource folder which belongs to the dao module
+```
+    spring.datasource.driver-class-name=com.mysql.jdbc.Driver
+    spring.datasource.url=jdbc:mysql://localhost:3306/dolphinscheduler
+    spring.datasource.username=ds_user
+    spring.datasource.password=dolphinscheduler  
+```
+
+3. Modify the pom.xml under the root directory, change the scope of the mysql-connector-java to complie
```
Comment (Segun-Ogundipe) [inline docs/en-us/development/development-environment-setup.md:80]: 3. Modify pom.xml in the root directory and modify the scope of mysql-connector-java to compile

File: docs/en-us/development/development-environment-setup.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+4. Refresh the dao module  
+try to run the main method inside org.apache.dolphinscheduler.dao.upgrade.shell.CreateDolphinScheduler to insert tables and data needed for the project.
```
Comment (Segun-Ogundipe) [inline docs/en-us/development/development-environment-setup.md:83]: 4. Refresh the dao module, run the main method of org.apache.dolphinscheduler.dao.upgrade.shell.CreateDolphinScheduler to automatically insert the tables and data required by the project

File: docs/en-us/development/development-environment-setup.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+5. Modify the service module  
+try to change the zookeeper.quorum part of the zookeeper.properties file
+zookeeper.quorum=localhost:2181
+
+6. Modify the dolphinscheduler-ui module  
+try to change the .env file
```
Comment (Segun-Ogundipe) [inline docs/en-us/development/development-environment-setup.md:90]: 5. Modify the .env file of the dolphinscheduler-ui module

File: docs/en-us/development/development-environment-setup.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    DEV_HOST = localhost
+```
+
+#### Start the project
+
+1. Start zookeeper  
+./bin/zkServer.sh start
+
+2. Start MasterServer  
+try to run the main method inside org.apache.dolphinscheduler.server.master.MasterServer and VM Options is needed:
```
Comment (Segun-Ogundipe) [inline docs/en-us/development/development-environment-setup.md:102]: 2. Start MasterServer
run the main method of org.apache.dolphinscheduler.server.master.MasterServer, you need to set the following VM options:

File: docs/en-us/development/development-environment-setup.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+```
+    -Dlogging.config=classpath:logback-master.xml -Ddruid.mysql.usePingMethod=false
+```
+3. Start WorkerServer
```
Comment (Segun-Ogundipe) [inline docs/en-us/development/development-environment-setup.md:106]: 3. Start WorkerServer
run the main method of org.apache.dolphinscheduler.server.worker.WorkerServer, you need to set the following VM options:

File: docs/en-us/development/development-environment-setup.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+```
+    -Dlogging.config=classpath:logback-master.xml -Ddruid.mysql.usePingMethod=false
+```
+3. Start WorkerServer  
+try to run the main method inside org.apache.dolphinscheduler.server.worker.WorkerServer and VM Options is needed:
+```
+    -Dlogging.config=classpath:logback-worker.xml -Ddruid.mysql.usePingMethod=false
+```
+4. Start ApiApplicationServer  
+try to run the main method inside org.apache.dolphinscheduler.api.ApiApplicationServer and VM Options is needed:
```
Comment (Segun-Ogundipe) [inline docs/en-us/development/development-environment-setup.md:112]: 4. Start ApiApplicationServer
run the main method of org.apache.dolphinscheduler.api.ApiApplicationServer, you need to set the following VM options:

File: docs/en-us/development/development-environment-setup.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+    else
+      echo "Error: No command named \`$command' was found."
+      exit 1
+    fi
+```
+
+6. Start the front-end ui modules  
+cd dolphinscheduler-ui directory
+And run 
+npm run start
```
Comment (Segun-Ogundipe) [inline docs/en-us/development/development-environment-setup.md:141]: cd dolphinscheduler-ui directory and run `npm run start`

File: docs/en-us/development/development-environment-setup.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+#### Visit the project
+1. Visit http://localhost:8888
+
+2. Sign in with the administrator role
```
Comment (Segun-Ogundipe) [inline docs/en-us/development/development-environment-setup.md:146]: 2. Sign in with the administrator account

File: md_json/docs.json [hunk 11]
```diff
@@ -30,6 +30,11 @@
       "link": "/en-us/docs/development/developers.html",
       "meta": {}
     },
+    {
+      "filename": "development-environment-setup.md",
+      "link": "/en-us/docs/development/development-environment-setup.html",
+      "meta": {}
+    },
```
Comment (Segun-Ogundipe) [inline md_json/docs.json:37]: You shouldn't edit this file, it is automatically generated and modified

**Review by Segun-Ogundipe — CHANGES_REQUESTED** [review]
Good job on this!!!
I have left some comments.

I see you added translation for the DS-License.md file too. You should create another PR to address that, this PR should only be for the environment setup translation. Also, try to change where you used `folder` to `directory`.

File: docs/en-us/development/development-environment-setup.md [hunk 12]
```diff
@@ -0,0 +1,148 @@
+#### Preparation
+
+1. First, fork the dolphinscheduler code from the remote repository to your local repository.
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/development/development-environment-setup.md:3]: have added link to the repo

File: docs/en-us/development/development-environment-setup.md [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+2. Install MySQL/PostgreSQL, JDK and MAVEN in your own software development environment.
+
+3. Clone your forked repository to the local file system.
+```
+    git clone https://github.com/apache/incubator-dolphinscheduler.git`
+```
+4. After finished the clone, go into the project folder and execute the following commands:
+```
+    git branch -a  #check the branch
+    git checkout dev #shift the dev branch
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/development/development-environment-setup.md:14]: done changes

File: docs/en-us/development/development-environment-setup.md [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+    git pull #sychronize the branch
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/development/development-environment-setup.md:15]: done changes

File: docs/en-us/development/development-environment-setup.md [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+    mvn -U clean package -Prelease -Dmaven.test.skip=true  #need to compile the necessary classes due to the project use gRPC dependencies
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/development/development-environment-setup.md:16]: done changes

File: docs/en-us/development/development-environment-setup.md (hunk 1 above)
Comment (Tianqi-Dotes) [reply docs/en-us/development/development-environment-setup.md:72]: done changes

File: docs/en-us/development/development-environment-setup.md (hunk 2 above)
Comment (Tianqi-Dotes) [reply docs/en-us/development/development-environment-setup.md:65]: done changes

File: docs/en-us/development/development-environment-setup.md (hunk 3 above)
Comment (Tianqi-Dotes) [reply docs/en-us/development/development-environment-setup.md:80]: done changes

File: docs/en-us/development/development-environment-setup.md (hunk 4 above)
Comment (Tianqi-Dotes) [reply docs/en-us/development/development-environment-setup.md:83]: done changes

File: docs/en-us/development/development-environment-setup.md (hunk 5 above)
Comment (Tianqi-Dotes) [reply docs/en-us/development/development-environment-setup.md:90]: done changes

File: docs/en-us/development/development-environment-setup.md (hunk 6 above)
Comment (Tianqi-Dotes) [reply docs/en-us/development/development-environment-setup.md:102]: done changes

File: docs/en-us/development/development-environment-setup.md (hunk 7 above)
Comment (Tianqi-Dotes) [reply docs/en-us/development/development-environment-setup.md:106]: done changes

File: docs/en-us/development/development-environment-setup.md (hunk 8 above)
Comment (Tianqi-Dotes) [reply docs/en-us/development/development-environment-setup.md:112]: done changes

File: docs/en-us/development/development-environment-setup.md (hunk 9 above)
Comment (Tianqi-Dotes) [reply docs/en-us/development/development-environment-setup.md:141]: done changes

File: docs/en-us/development/development-environment-setup.md (hunk 10 above)
Comment (Tianqi-Dotes) [reply docs/en-us/development/development-environment-setup.md:146]: done changes

**Comment by Tianqi-Dotes** [comment]
@Segun-Ogundipe 
hi Segun,
Have done the changes according to ur suggestions, deep thanx for spent much time on the proofreading.
Sincerely
Tianqi

File: md_json/docs.json (hunk 11 above)
Comment (Tianqi-Dotes) [reply md_json/docs.json]: done deletion

**Review by Segun-Ogundipe — APPROVED** [review]

**Review by dailidong — APPROVED** [review]
+1
good job

**Comment by dailidong** [comment]
@Tianqi-Dotes ，thanks for your first contribution, please add my wechat: 510570367 or mail: dailidong66@gmail.com, when added, please tell me your github id, thx

**Pull request merged**